packages/compiler/src/parser.js

```diff
@@ -62,6 +71,16 @@ export class Parser {
       }
     }
     if (!vars.length) return ''
-    return 'const ' + ["s = String(n).split('.')", ...vars].join(', ')
+    if (this.c) {
+      vars.unshift(
+        'se = String(n).split(/[ce]/)',
```

Comment:
```suggestion
        "se = String(n).split('c')",
```
If/once CLDR-14420 is resolved to clarify that scientific notation values should not get an `e` operand value, this change needs to be made.